src/live.rs

```diff
@@ -768,7 +772,146 @@ fn copy_file_from_iso(iso: &mut IsoFs, file: &iso9660::File, output_path: &Path)
+}
+
+pub fn iso_pack_minimal_iso(config: &IsoExtractPackMinimalIsoConfig) -> Result<()> {
+    let mut full_iso = IsoFs::from_file(open_live_iso(&config.full, Some(None))?)?;
+    let mut minimal_iso = IsoFs::from_file(open_live_iso(&config.minimal, None)?)?;
+
+    let full_files = collect_iso_files(&mut full_iso)
+        .with_context(|| format!("collecting files from {}", &config.full))?;
+    let minimal_files = collect_iso_files(&mut minimal_iso)
+        .with_context(|| format!("collecting files from {}", &config.minimal))?;
+    if full_files.is_empty() {
+        bail!("No files found in {}", &config.full);
+    } else if minimal_files.is_empty() {
+        bail!("No files found in {}", &config.minimal);
+    }
+
+    let table = miniso::Table::new(&full_files, &minimal_files).context("building table")?;
+    eprintln!("Matched {} files of {}", table.len(), minimal_files.len());
+
+    eprintln!("Packing minimal ISO");
+    let (data, skipped, written, written_compressed) = table
+        .xzpack(minimal_iso.as_file()?)
+        .context("packing miniso")?;
+
+    eprintln!("Total bytes skipped: {}", skipped);
+    eprintln!("Total bytes written: {}", written);
+    eprintln!("Total bytes written (compressed): {}", written_compressed);
+
+    eprintln!("Verifying that packed image matches digest");
+    data.unxzpack(full_iso.as_file()?, &std::env::temp_dir())
+        .context("unpacking miniso for verification")?;
+
+    let miniso_entry = full_iso
+        .get_path(COREOS_ISO_MINISO_FILE)
+        .with_context(|| format!("looking up '{}'", COREOS_ISO_MINISO_FILE))?
+        .try_into_file()?;
+    let mut w = full_iso.overwrite_file(&miniso_entry)?;
+    data.serialize(&mut w).context("writing miniso data file")?;
+    w.flush().context("flushing full ISO")?;
+
+    if config.consume {
+        std::fs::remove_file(&config.minimal)
+            .with_context(|| format!("consuming {}", &config.minimal))?;
+    }
+
+    eprintln!("Packing successful!");
+    Ok(())
+}
+
+fn collect_iso_files(iso: &mut IsoFs) -> Result<HashMap<String, iso9660::File>> {
+    iso.walk()?
+        .filter_map(|r| match r {
+            Err(e) => Some(Err(e)),
+            Ok((s, iso9660::DirectoryRecord::File(f))) => Some(Ok((s, f))),
+            Ok(_) => None,
+        })
+        .collect::<Result<HashMap<String, iso9660::File>>>()
+        .context("while walking ISO filesystem")
+}
+
+fn modify_miniso_kargs(f: &mut File, rootfs_url: Option<&String>) -> Result<()> {
+    let mut iso = IsoConfig::for_file(f)?;
+
+    let kargs = iso.kargs()?;
+
+    // same disclaimer as `modify_kargs()` here re. whitespace/quoting
+    let liveiso_karg = kargs
+        .split_ascii_whitespace()
+        .find(|&karg| karg.starts_with("coreos.liveiso="))
+        .ok_or_else(|| anyhow!("minimal ISO does not have coreos.liveiso= karg"))?
+        .to_string();
+
+    let rootfs_url_karg = rootfs_url.map(|url| vec![format!("coreos.live.rootfs_url={}", url)]);
```

Comment:
I don't know if the URL might possibly be supplied across a trust boundary, but if so, it can be used here for karg injection.

Reply:
Yeah... I guess we could at least do a whitespace check here? In theory we could just double-quote it (after verifying that there are no double-quotes inside), but a lot of our code (and I'm sure other people's code) doesn't parse it correctly.